internal/uploader/auth_utils.go

```diff
@@ -0,0 +1,96 @@
+package uploader
+
+import (
+	"bytes"
+	"encoding/base64"
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/go-openapi/strfmt"
+	"github.com/openshift/assisted-service/internal/common"
+	"github.com/openshift/assisted-service/pkg/k8sclient"
+	"github.com/pkg/errors"
+	log "github.com/sirupsen/logrus"
+	"gorm.io/gorm"
+	corev1 "k8s.io/api/core/v1"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	openshiftTokenKey = "cloud.openshift.com"
+)
+
+type serializedAuthMap struct {
+	Auths map[string]serializedAuth `json:"auths"`
+}
+type serializedAuth struct {
+	Auth  string `json:"auth"`
+	Email string `json:"email,omitempty"`
+}
+
+func getPullSecret(db *gorm.DB, clusterID *strfmt.UUID, k8sclient k8sclient.K8SClient, key string) (*serializedAuth, error) {
+	data, exists := getOCMPullSecret(k8sclient)
+	if !exists {
+		data = getTokenFromCluster(db, clusterID)
+	}
+	var pullSecret serializedAuthMap
+	if err := json.Unmarshal(data, &pullSecret); err != nil {
+		return nil, errors.Wrapf(err, "failed to unmarshal pull secret")
+	}
+	if auth, ok := pullSecret.Auths[key]; ok {
+		return &auth, nil
+	}
+	return nil, errors.Errorf("pull secret doesn't contain authentication information for %s", key)
+}
+
+func getOCMPullSecret(k8sclient k8sclient.K8SClient) ([]byte, bool) {
+	if k8sclient != nil {
+		secret, err := k8sclient.GetSecret("openshift-config", "pull-secret")
+		if err != nil {
+			err = client.IgnoreNotFound(err)
+			// A "not found" error (err == nil) should return false
+			// indicating that the secret does not exist
+			return nil, err != nil
+		}
+		return secret.Data[corev1.DockerConfigJsonKey], true
+	}
+	return nil, false
+}
+
+func getTokenFromCluster(db *gorm.DB, clusterID *strfmt.UUID) []byte {
+	cluster, err := common.GetClusterFromDB(db, *clusterID, common.SkipEagerLoading)
+	if err != nil {
+		log.WithError(err).Errorf("failed to find cluster in db")
```

Comment:
pls add cluster-ID to this message for debuggability

Reply:
Sorry this log was for my own testing; I've removed it since I thought the error returned here doesn't matter for finding the cluster's pull secret. 

Do you think it's better to have a debug log here, though?

Reply:
this one is no longer relevant since the cluster entity is passed to this function.